sycl/include/sycl/builtins.hpp

```diff
@@ -521,7 +521,14 @@ detail::enable_if_t<
     detail::is_svgenfloat<T>::value && detail::is_genfloatptr<T2>::value, T>
 fract(T x, T2 iptr) __NOEXC {
   detail::check_vector_size<T, T2>();
-  return __sycl_std::__invoke_fract<T>(x, iptr);
+  // workaround to support multi_ptr without decoration
+  if constexpr (!iptr.is_decorated) {
+    multi_ptr<typename detail::remove_pointer_t<T2>, T2::address_space,
```

Comment:
Tag @steffenlarsen in case you have better solution

Reply:
Steffen proposed to use `get_decorated()` here 

https://github.com/intel/llvm/blob/09ceb1034f04ef7156c7ea2be0f3cd1bcbc12df2/sycl/include/sycl/detail/generic_type_traits.hpp#L391 

for pointers `t` with `decorated::no`. Will try tomorrow. It will convert non-decorated pointer to decorated  - the same logic as it is now in this PR, but the implementation should look much cleaner.

Reply:
But it would be plain pointer (`__global int *`), not `multi_ptr`. What am I missing?

Reply:
@aelovikov-intel, the built-in accepts `genfloatptr` in accordance with the SYCL spec, which is in turn defined as `multir_ptr` in [Table 165](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.gentypes)

> genfloatptr
> All permutations of multi_ptr<DataT, AddressSpace, IsDecorated> where DataT is all types within genfloat, AddressSpace is access::address_space::global_space, access::address_space::local_space and access::address_space::private_space and IsDecorated is access::decorated::yes and access::decorated::no.

Essentially, you can't pass a raw pointer to the built-in. But I would double-check if our definition of `is_genfloatptr` matches the spec

Reply:
> `is_genfloatptr`

Correct, it doesn't match the spec. I keep in mind that the separate patch should be made which deprecates current scalar functions which accept decorated::legacy, but this is second priority, so didn't introduce it in this PR.